### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by pykello** [citusdata/citus#3468]
Using the master branch.

```sql
create table source_table(a int, b int, c int);
SET citus.shard_count TO 3;
SELECT create_distributed_table('source_table', 'a');

CREATE SEQUENCE serial START 101;

select (select count(*) from source_table ), nextval('serial');
```

```
#0  RebuildQueryStrings (originalQuery=originalQuery@entry=0x55cd59be2ad8, 
    taskList=0x55cd59be3918) at planner/deparse_shard_query.c:52
#1  0x00007f68d662c1b8 in CitusBeginScanWithCoordinatorProcessing (estate=<optimized out>, 
    eflags=<optimized out>, node=0x55cd59b0a8b0) at executor/citus_custom_scan.c:270
#2  CitusBeginScan (node=0x55cd59b0a8b0, estate=<optimized out>, eflags=<optimized out>)
    at executor/citus_custom_scan.c:152
#3  0x000055cd58406387 in ExecInitCustomScan (cscan=cscan@entry=0x55cd59baf458, 
    estate=estate@entry=0x55cd59b0a698, eflags=eflags@entry=16) at nodeCustom.c:101
#4  0x000055cd583f6b3f in ExecInitNode (node=node@entry=0x55cd59baf458, 
    estate=estate@entry=0x55cd59b0a698, eflags=eflags@entry=16) at execProcnode.c:282
#5  0x000055cd583f00ff in InitPlan (eflags=16, queryDesc=0x55cd59baf458) at execMain.c:1046
#6  standard_ExecutorStart (queryDesc=queryDesc@entry=0x55cd59ab37d8, eflags=16, 
    eflags@entry=0) at execMain.c:265
#7  0x00007f68d663135a in CitusExecutorStart (queryDesc=0x55cd59ab37d8, eflags=0)
    at executor/multi_executor.c:111
#8  0x000055cd58558166 in PortalStart (portal=portal@entry=0x55cd59a71118, 
    params=params@entry=0x0, eflags=eflags@entry=0, snapshot=snapshot@entry=0x0)
    at pquery.c:520
#9  0x000055cd5855415d in exec_simple_query (
    query_string=0x55cd5999b1c8 "select (select count(*) from source_table ), nextval('serial');") at postgres.c:1106
```

**Comment by pykello** [comment]
I was using the wrong installation. Seems to work on master. closing.

**Comment by onderkalaci** [comment]
For future reference, introduced by #3388  fixed by #3440. And, this issue looks the exact same with https://github.com/citusdata/citus/issues/3436
